### PR TITLE
Generate azure_resource_id

### PR DESCRIPTION
**Pull request by pmcollins**
When a resource detection processor is running and we're in an Azure environment, we can use the
Azure resource attributes to generate an `azure_resource_id`, which is used by the SFx backend.

This functionality aims to provide parity with the SFx Smart Agent's generation of `azure_resource_id`.

Some code taken from https://github.com/signalfx/signalfx-agent/blob/master/pkg/core/hostid/azure.go#L18

Tested on Azure (standalone VM and Scale Set).

### CODE REVIEW

File: internal/splunk/hostid.go [hunk 1]
```diff
@@ -92,3 +112,43 @@ func ResourceToHostID(res pdata.Resource) (HostID, bool) {
 
 	return HostID{}, false
 }
+
+func azureID(attrs pdata.AttributeMap, cloudAccount string) string {
+	var resourceGroupName string
+	if attr, ok := attrs.Get("azure.resourcegroup.name"); ok {
```
Comment (asuresh4) [inline internal/splunk/hostid.go:118]: Would it make sense to make `azure.resourcegroup.name` a constant that can be accessed both by the processor and this package? Same thing with `azure.vm.scaleset.name`.

**Review by asuresh4 — APPROVED** [review]
lgtm, apart from the one comment.

File: internal/splunk/hostid.go (hunk 1 above)
Comment (pmcollins) [reply internal/splunk/hostid.go:118]: Yeah I think so. There's also a comment Bogdan made about possibly adding these to the semantic conventions.
https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/2890#issuecomment-808350732
Will look into it.

Comment (asuresh4) [reply internal/splunk/hostid.go:118]: sgtm

**Review by jrcamp — APPROVED** [review]

**Comment by bd-clara** [comment]
Hi,

This is wrong, azure resource name and hostname can be different

Actually it generates on some of my VM an ID with localhost inside
`azure_resource_id:xxxxxxxxxxxxxxxxxxxxxxxxxxxx/xxxxxx-xxx-xxxx-shared-services-support-rg/microsoft.compute/virtualmachines/localhost`

Azure Resource ID is part of the VM metadata
`curl -s -H Metadata:true --noproxy "*" "http://169.254.169.254/metadata/instance?api-version=2021-02-01"`
Two solution, `.compute.name` or `.compute.resourceId`

Best regards,

**Pull request merged**